src/main/application-update/periodical-check-for-updates/periodical-check-for-updates.injectable.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { getStartableStoppable } from "../../../common/utils/get-startable-stoppable";
+import processCheckingForUpdatesInjectable from "../check-for-updates/process-checking-for-updates.injectable";
+
+const periodicalCheckForUpdatesInjectable = getInjectable({
+  id: "periodical-check-for-updates",
+
+  instantiate: (di) => {
+    const processCheckingForUpdates = di.inject(processCheckingForUpdatesInjectable);
+
+    return getStartableStoppable("periodical-check-for-updates", () => {
+      const TWO_HOURS = 1000 * 60 * 60 * 2;
+
+      // Note: intentional orphan promise to make checking for updates happen in the background
+      processCheckingForUpdates();
+
+      const intervalId = setInterval(() => {
+        // Note: intentional orphan promise to make checking for updates happen in the background
+        processCheckingForUpdates();
+      }, TWO_HOURS);
```

Comment:
Yeah 2 hours is probably longer than most timeouts, but shouldn't we close the old one just in case?

Reply:
This being `startableStoppable`, there is no way to for multiple timeouts to exist if that's what you mean :)

Reply:
Not multiple timeouts, multiple in-flight requests. If `processCheckingForUpdates` takes more than 2 hours then there would be multiple.

Not very important, but something to keep in mind I think.

Reply:
Hmm. I'd say that in that case we shouldn't start the new check